### PR TITLE
Port changes of Maestro's Incoming page to PCS

### PR DESCRIPTION
**Pull request by premun**
Porting https://github.com/dotnet/arcade-services/issues/3869

### CODE REVIEW

File: src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace ProductConstructionService.Api.Pages.DependencyFlow;
 
 public class IncomingModel : PageModel
 {
-    private static readonly Regex _repoParser = new(@"https?://(www\.)?github.com/(?<owner>[A-Za-z0-9-_\.]+)/(?<repo>[A-Za-z0-9-_\.]+)");
+    private static readonly Regex _repoParser = new Regex(@"https?://(www\.)?github.com/(?<owner>[A-Za-z0-9-_\.]+)/(?<repo>[A-Za-z0-9-_\.]+)");
```
Comment (oleksandr-didyk) [inline src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:19]: Is there a difference between `new()` and `new Regex()` in this context?

Comment (dkurepa) [inline src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:19]: does this not work for some reason?

File: src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs [hunk 2]
```diff
@@ -30,7 +30,7 @@ public IncomingModel(
     {
         _context = context;
         // We'll only comparing public commits, so we don't need a token.
-        _github = gitHubClientFactory.CreateGitHubClient(string.Empty);
+        _github = gitHubClientFactory.CreateGitHubClient("");
```
Comment (dkurepa) [inline src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:33]: why change this?

**Review by oleksandr-didyk — DISMISSED** [review]

File: src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs (hunk 2 above)
Comment (premun) [reply src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:33]: I copied the file from Maestro

File: src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs (hunk 1 above)
Comment (premun) [reply src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:19]: No, I just copied the file

Comment (premun) [reply src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:19]: I copied this, we can revert the change

```suggestion
    private static readonly Regex _repoParser = new(@"https?://(www\.)?github.com/(?<owner>[A-Za-z0-9-_\.]+)/(?<repo>[A-Za-z0-9-_\.]+)");
```

File: src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs (hunk 2 above)
Comment (premun) [reply src/ProductConstructionService/ProductConstructionService.Api/Pages/DependencyFlow/Incoming.cshtml.cs:33]:
```suggestion
        _github = gitHubClientFactory.CreateGitHubClient(string.Empty);
```

**Review by oleksandr-didyk — APPROVED** [review]

**Review by dkurepa — APPROVED** [review]

**Pull request merged**